javaparser-core/src/main/java/com/github/javaparser/ast/comments/MarkdownComment.java

```
@@ -0,0 +1,211 @@
+     */
+    @Generated("com.github.javaparser.generator.core.node.MainConstructorGenerator")
+    public MarkdownComment(TokenRange tokenRange, String content) {
+        super(tokenRange, content);
+        customInitialization();
+    }
+
+    /**
+     * Returns the Markdown content of this comment as defined in <a href="https://openjdk.org/jeps/467">JEP 467</a>:
+     * <blockquote cite="https://openjdk.org/jeps/467">
+     *     Because horizontal whitespace at the beginning and end of each line of Markdown text may be significant,
+     *     the content of a Markdown documentation comment is determined as follows:
+     *     -- Any leading whitespace and the three initial / characters are removed from each line.
+     *     -- The lines are shifted left, by removing leading whitespace characters, until the non-blank line with the
+     *        least leading whitespace has no remaining leading whitespace.
+     *     -- Additional leading whitespace and any trailing whitespace in each line is preserved, because it may be
+     *        significant. For example, whitespace at the beginning of a line may indicate an indented code block or the
+     *        continuation of a list item, and whitespace at the end of a line may indicate a hard line break.
+     *     </blockquote>
+     * @return
+     */
+    public String getMarkdownContent() {
+        String content = getContent();
+        LineSeparator lineSeparator = LineSeparator.detect(content);
+        ArrayList<String> commentLines = new ArrayList<>();
+        if (lineSeparator == LineSeparator.CR
+                || lineSeparator == LineSeparator.LF
+                || lineSeparator == LineSeparator.CRLF) {
+            commentLines.addAll(Arrays.asList(content.split(lineSeparator.asRawString())));
+        } else if (lineSeparator == LineSeparator.NONE) {
+            commentLines.add(content);
+        } else {
+            // TODO
```

Maybe it would be better to throw an exception in this case.

---

I think this whole if/else chain can be simplified to splitting on a single regex that covers all the cases.